prow/github/hmac.go

```diff
@@ -20,11 +20,22 @@ import (
 	"crypto/hmac"
 	"crypto/sha1"
 	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"sigs.k8s.io/yaml"
 	"strings"
+	"time"
 )
 
 // ValidatePayload ensures that the request payload signature matches the key.
-func ValidatePayload(payload []byte, sig string, key []byte) bool {
+func ValidatePayload(payload []byte, sig string, tokenGenerator func() []byte) bool {
+	var event GenericEvent
+	if err := json.Unmarshal(payload, &event); err != nil {
+		fmt.Printf("error: %v", err)
```

Comment:
Use the logging library (`github.com/sirupsen/logrus`), and be specific about what errored.

```suggestion
		logrus.WithError(err).Info("ValidatePayload couldn't unmarshal the github event payload")
```